Tasks/NuGetInstaller/VsoNuGetHelper.ps1

```diff
@@ -0,0 +1,163 @@
+Add-Type -AssemblyName System.Security
+
+$nuGetTempDirectory = Join-Path $Env:AGENT_BUILDDIRECTORY "NuGet\"
+$tempNuGetConfigPath = Join-Path $nuGetTempDirectory "newNuGet.config"
+
+function SaveTempNuGetConfig
+{
+    [CmdletBinding()]
+    param(
+        [xml] $nugetConfig        
+    )
+
+    if(-not (Test-Path -Path ($nuGetTempDirectory)))
+    {
+        New-Item -Path $nuGetTempDirectory -ItemType Directory
+    }
+    Write-Output "Saving to $tempNuGetConfigPath"
```

Comment:
It looks like you meant to Write-Host here, not Write-Output. Write-Output yields the object as a result of the function call, so if no one intercepts the value it ends up getting written to the host. But if it's intercepted and discarded by the caller, then it never makes it to the host. At minimum this is confusing.